### PR TITLE
Split current bump PR into 2: one for jobs and one for prow object files.

### LINKED ISSUE #2550: Split current bump PR into 2: one for jobs and one for prow object files.
Separation of concerns so Prow oncall can bump Prow-specific components without the approval of Istio T&R and vice versa wrt jobs.

PR in question: https://github.com/istio/test-infra/pull/2538

If all the fields are customizable in `autobump` (e.g. title, dirs, etc.), then we can probably just use the script twice with different parameters, otherwise, Automator can be used for the job-specific and miscellaneous other image bumps.

### PR DESCRIPTION
**Pull request by clarketm**
fix https://github.com/istio/test-infra/issues/2550

### CODE REVIEW

File: prow/cluster/jobs/istio/test-infra/istio.test-infra.trusted.master.yaml
Comment (clarketm) [inline prow/cluster/jobs/istio/test-infra/istio.test-infra.trusted.master.yaml:98]: This needs rework or more support in autobump.

File: prow/cluster/jobs/istio/test-infra/istio.test-infra.trusted.master.yaml
Comment (cjwagner) [inline prow/cluster/jobs/istio/test-infra/istio.test-infra.trusted.master.yaml]: We need to bump this image with the rest of the prow components:
https://github.com/istio/test-infra/blob/89fa0e1fae5964c33780c59583b785a05c2e0d91/.prow.yaml#L8

File: prow/config/jobs/test-infra.yaml
Comment (cjwagner) [inline prow/config/jobs/test-infra.yaml]: Doesn't this need to exclude `prow/cluster/jobs/istio/test-infra`?

File: prow/cluster/jobs/istio/test-infra/istio.test-infra.trusted.master.yaml
Comment (clarketm) [reply prow/cluster/jobs/istio/test-infra/istio.test-infra.trusted.master.yaml]: I need to either change ownership of the file or move this job back to `.../jobs/istio/test-infra/` directory. I will do the former.

File: prow/config/jobs/test-infra.yaml
Comment (clarketm) [reply prow/config/jobs/test-infra.yaml]: Technically no, since the image version of plank (i.e. `--source`) should always be the same as the image versions of jobs in `test-infra/` (since they are bumped together by autobump).

It doesn't hurt to exclude it, but I will need to enable `extglob` first to ensure the syntax remains simple.

File: prow/cluster/jobs/istio/test-infra/istio.test-infra.trusted.master.yaml
Comment (clarketm) [reply prow/cluster/jobs/istio/test-infra/istio.test-infra.trusted.master.yaml]: Updated code and modified CODEOWNERS.

File: prow/config/jobs/test-infra.yaml
Comment (cjwagner) [reply prow/config/jobs/test-infra.yaml]: Ah so that should always be a noop. 👍 
Excluding might be preferable, but it doesn't seem important and shouldn't hold this up.

**Review by cjwagner — APPROVED** [review]

File: prow/config/jobs/test-infra.yaml
Comment (clarketm) [reply prow/config/jobs/test-infra.yaml]: I went ahead and added an exclusion on `test-infra` to tidy this up (and ensure this doesn't regress  in case the logic changes in the future)

**Review by cjwagner — APPROVED** [review]
LGTM, but I don't think I can approve fully.

**Review by howardjohn — APPROVED** [review]

**Pull request merged**
